src/js/stores/document.js

```diff
@@ -169,7 +169,17 @@ define(function (require, exports, module) {
 
             this._openDocuments[nextDocument.id] = nextDocument;
 
-            if (!suppressChange) {
+            if (suppressChange) {
+                return;
+            }
+
+            // If some selected layer remains uninitialized, a new document will
+            // come along shortly. Wait until then to trigger the change event.
+            var initialized = nextDocument.layers.selected.every(function (layer) {
```

Comment:
I understand your concern, but it was intentional. The problem is that it's very difficult to ensure that we don't emit a change event between the time when the selection has changed and when the properties have been initialized. If we don't limit the change events here, then we get a) a small number of errors in the JSX due to accessing uninitialized properties (which isn't a big deal) and b) more importantly, a lot of blinking in the UI due to drawing panels without initialized properties. If I remove this, there has to be a lot more `shouldComponentUpdate` code overhead. I decided instead to do it here to codify an assumption that the UI can make: if it's selected, then it's initialized. It solves all the above problems in two lines of code. Still, I can see why you wouldn't like it and I'm happy to discuss it further.